contracts/rewards/src/state.rs

```diff
@@ -33,13 +32,36 @@ impl EpochTally {
             epoch,
         }
     }
+
+    pub fn record_participation(&mut self, worker: Addr) -> &Self {
```

Comment:
```suggestion
    pub fn record_participation(mut self, worker: Addr) -> Self {
```
You can avoid the extra `.clone` by consuming self.

Reply:
yeah but then we need to do:
```
let tally = tally.record_participation(...);
```
instead of:
```
tally.record_participation(...);
```

I prefer the latter. Also, the compiler will optimize away the extra clone. But we should converge on a style.